bee-ledger/src/workers/snapshot/download.rs

```diff
@@ -21,23 +21,30 @@ async fn download_snapshot_header(download_url: &str) -> Result<SnapshotHeader,
 
     match reqwest::get(download_url).await.and_then(Response::error_for_status) {
         Ok(res) => {
-            let mut stream = res.bytes_stream();
-            let mut bytes = Vec::<u8>::with_capacity(SnapshotHeader::LENGTH);
+            if res.status().is_success() {
+                let mut stream = res.bytes_stream();
+                let mut bytes = Vec::<u8>::with_capacity(SnapshotHeader::LENGTH);
 
-            while let Some(chunk) = stream.next().await {
-                let mut chunk_reader = chunk.map_err(|_| Error::DownloadingFailed)?.reader();
+                while let Some(chunk) = stream.next().await {
+                    let mut chunk_reader = chunk.map_err(|_| Error::DownloadingFailed)?.reader();
 
-                let mut buf = Vec::new();
-                chunk_reader.read_to_end(&mut buf)?;
-                bytes.extend_from_slice(&buf);
+                    let mut buf = Vec::new();
+                    chunk_reader.read_to_end(&mut buf)?;
+                    bytes.extend_from_slice(&buf);
 
-                if bytes.len() >= SnapshotHeader::LENGTH {
-                    debug!("Downloaded snapshot header from {}.", download_url);
+                    if bytes.len() >= SnapshotHeader::LENGTH {
+                        debug!("Downloaded snapshot header from {}.", download_url);
 
-                    let mut slice: &[u8] = &bytes[..SnapshotHeader::LENGTH];
+                        let mut slice: &[u8] = &bytes[..SnapshotHeader::LENGTH];
 
-                    return Ok(SnapshotHeader::unpack(&mut slice)?);
+                        return Ok(SnapshotHeader::unpack(&mut slice)?);
+                    }
                 }
+            } else {
+                debug!(
+                    "Downloading snapshot header failed with status code {:?}.",
+                    res.status()
+                );
```

Comment:
Should this be a `warn!(...)`, to be consistent with the warning at line 171?

Reply:
I think we decided on using `debug` for the headers so that we don't spam the log too much.